frontend/src/features/modules/ModulesTree.tsx

```diff
@@ -1,60 +1,152 @@
 import { Disclosure, DisclosureButton, DisclosurePanel } from '@headlessui/react'
-import { ChevronRightIcon } from '@heroicons/react/24/outline'
+import {
+  ArrowRightCircleIcon,
+  BellAlertIcon,
+  BellIcon,
+  BoltIcon,
+  BookOpenIcon,
+  ChevronRightIcon,
+  CircleStackIcon,
+  CodeBracketSquareIcon,
+  Cog6ToothIcon,
+  DocumentDuplicateIcon,
+  LockClosedIcon,
+  NumberedListIcon,
+  SquaresPlusIcon,
+} from '@heroicons/react/24/outline'
+import { TableCellsIcon } from '@heroicons/react/24/solid'
+import type { ForwardRefExoticComponent, SVGProps } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import type { Decl } from '../../protos/xyz/block/ftl/v1/schema/schema_pb'
 import { classNames } from '../../utils'
 import type { ModuleTreeItem } from './module.utils'
 
+// This could alternatively be an icon, but we'd need to pick a good one.
+const ExportBadge = () => <span className='text-xs py-0.5 px-1.5 bg-gray-200 dark:bg-gray-900 dark:text-gray-300 rounded-md'>Exported</span>
+
+type IconMap = Record<string, ForwardRefExoticComponent<SVGProps<SVGSVGElement> & { title?: string; titleId?: string }>>
+const icons: IconMap = {
+  config: Cog6ToothIcon,
+  data: TableCellsIcon,
+  database: CircleStackIcon,
+  enum: NumberedListIcon,
+  fsm: SquaresPlusIcon,
+  topic: BellIcon,
+  typeAlias: DocumentDuplicateIcon,
+  secret: LockClosedIcon,
+  subscription: BellAlertIcon,
+  verb: BoltIcon,
+}
+
+type WithExport = { export?: boolean }
+
+const DeclNode = ({ decl, href }: { decl: Decl; href: string }) => {
```

Comment:
Should we `memo` this to avoid rebuilds?

Reply:
Changed a ton of stuff around here but I think I memoized everything that doesn't need to be rerendered